src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs

```diff
@@ -2048,18 +2047,24 @@ private void ParseExpectedErrorConfigurations()
 
             foreach (var errorClass in _localConfiguration.errorCollector.expectedMessages)
             {
-                var messages = errorClass.message;
-                if (messages != null)
+                if (errorClass.message != null)
                 {
-                    localExpectedErrorMessages.Add(errorClass.name, messages);
+                    localExpectedErrorMessages[errorClass.name] = errorClass.message;
                 }
             }
 
-            var expectedErrorInfo = ServerOverrides(_serverConfiguration.RpmConfig.ErrorCollectorExpectedMessages, localExpectedErrorMessages).ToDictionary();
+            var expectedErrorInfo = localExpectedErrorMessages;
```

Comment:
We should reverse the order of this.
Read the server side messages in first.  If there aren't any or it's empty, then we should process the local config.

This entire block of code should be performed after attempting to read the server config settings.